### ISSUE TITLE
tensorflow-gpu is downgraded to 1.7 when using conda update --all

### ISSUE THREAD

**Issue by charlesztt** [ContinuumIO/anaconda-issues#9698]
### Actual Behavior
When I am using
`conda update --all`
tensorflow-gpu (latest version on Linux is 1.8) will be downgraded to 1.7.

### Expected Behavior
Since I already have the latest version of tensorflow-gpu, it should not be downgraded.

### Steps to Reproduce (and my guess)
I start with a clean environment
`conda create -n issue-repro python=3 tensorflow-gpu`
This will install tensorflow-gpu for me
Then I use
`conda update --all`
 
I will see this.
```
Solving environment: done

## Package Plan ##

  environment location: /<hidden path>


The following NEW packages will be INSTALLED:

    tensorflow-gpu-base: 1.7.0-py36hcdda91b_1

The following packages will be UPDATED:

    _tflow_180_select:   1.0-gpu              --> 3.0-eigen           

The following packages will be DOWNGRADED:

    tensorboard:         1.8.0-py36hf484d3e_0 --> 1.7.0-py36hf484d3e_1
    tensorflow:          1.8.0-hb11d968_0     --> 1.7.0-0             
    tensorflow-base:     1.8.0-py36hc1a7637_0 --> 1.7.0-py36hdbcaa40_2
    tensorflow-gpu:      1.8.0-h7b35bdc_0     --> 1.7.0-0             

Proceed ([y]/n)? 

Preparing transaction: done
Verifying transaction: done
Executing transaction: done
```

I believe that the problem is caused by the package _tflow_180_select, tf-gpu 1.7 depends on 3.0-eigen, and tf-gpu 1.8 depends on 1.0-gpu. And conda think "3.0-eigen" > "1.0-gpu" and is newer, but the downstream packages are downgraded.

##### Anaconda or Miniconda version:
4.5.4

##### Operating System:
CentOS 7

**Comment by jjhelmus** [comment]
Yikes, this is not ideal, thanks for the excellent report @charlesztt .

The `_tflow_180_select` package is needed for making the non-gpu version of Tensorflow the default install option when a variant is not specified.  Specifically `conda install tensorflow` should install the CPU version of Tensorflow.  There is no reason that the version numbers of this package need to be 1.0, 2.0, and 3.0, smaller version differences are fine and should have less effect on the solver.  I don't think this can be fixed for Tensorflow 1.8.0 but I'll see what can be done for the upcoming 1.9.0 release.

**Comment by vazzup** [comment]
This bug still exists. In conda 4.5.11, with tensorflow-gpu-1.10.0 and _tflow_190_select, _tflow_190_select.

**Comment by jjhelmus** [comment]
@vazzup Can you provide more details on what is not working?  Upgrading from tensorflow-gpu 1.9.0 to 1.10.0 works fine using `conda update --all`.  The `_tflow_190_select` package is upgraded in this transaction but it has not effect on the environment as it is no longer needed.

**Comment by vazzup** [comment]
Inside my conda environment, running `conda upgrade --all` gives me the following output. Running conda 4.5.11 on Arch Linux.
![screenshot from 2018-09-21 13-21-22](https://user-images.githubusercontent.com/9087638/45867631-80051080-bda1-11e8-9f07-94ec52263d62.png)

**Comment by jjhelmus** [comment]
@vazzup I'm not able to replicate this myself.  Can you provide the output of `conda info` and `conda list --show-channel-urls`.  By best guess is that is some other package in the environment that is causing conda to prefer the 1.9.0 packages.

**Comment by vazzup** [comment]
@jjhelmus Sorry for the delay. Output of `conda info` is in the image. Output  of `conda list --show-channel-urls` can be found at https://pastebin.com/8rpvMzPY 
![screenshot from 2018-09-24 17-57-36](https://user-images.githubusercontent.com/9087638/45952176-c8336580-c023-11e8-92c2-2763aa84c991.png)

**Comment by jjhelmus** [comment]
@vazzup Thanks for the information.  I was able to replicate this behavior with the following on `linux-64`:

```
$ conda create -n test python=3.6 nccl tensorflow-gpu
...
    _tflow_1100_select: 0.0.1-gpu                 defaults
...
    tensorboard:        1.10.0-py36hf484d3e_0     defaults
    tensorflow:         1.10.0-gpu_py36h97a2126_0 defaults
    tensorflow-base:    1.10.0-gpu_py36h6ecc378_0 defaults
    tensorflow-gpu:     1.10.0-hf154084_0         defaults
...
$ conda update -n test --all
Solving environment: done

## Package Plan ##

  environment location: /home/jhelmus/anaconda3/envs/test


The following NEW packages will be INSTALLED:

    _tflow_190_select:  0.0.1-gpu                 defaults

The following packages will be UPDATED:

    _tflow_1100_select: 0.0.1-gpu                 defaults --> 0.0.3-mkl                defaults

The following packages will be DOWNGRADED:

    tensorboard:        1.10.0-py36hf484d3e_0     defaults --> 1.9.0-py36hf484d3e_0     defaults
    tensorflow:         1.10.0-gpu_py36h97a2126_0 defaults --> 1.9.0-gpu_py36h02c5d5e_1 defaults
    tensorflow-base:    1.10.0-gpu_py36h6ecc378_0 defaults --> 1.9.0-gpu_py36h6ecc378_0 defaults
    tensorflow-gpu:     1.10.0-hf154084_0         defaults --> 1.9.0-hf154084_0         defaults
```

@kalefranz @mcg1969 are ideas on how to adjust the tensorflow meta-packages or selection package to avoid this behavior?

**Comment by mcg1969** [comment]
Why are we allowing `_tflow_190_select` and `_tflow_1100_select` to live in the same environment? This seems problematic to me. Shouldn't we be having a single package `_tflow_select` that is versioned in parallel with tensorflow itself?

**Comment by vazzup** [comment]
@jjhelmus Yes that is the set of steps I had undertaken to set up conda.
